### ISSUE TITLE
Disabled transforms are breaking reports

### ISSUE THREAD

**Issue by rpl-will-hamill** [grafana/grafana#80947]
### What happened?

A panel is stating "No Data" when some transformations are disabled.
![image](https://github.com/grafana/grafana/assets/129032358/adc33e76-61ff-48b4-955c-73f87742b1f4)
![image](https://github.com/grafana/grafana/assets/129032358/03b49212-e21d-4b60-ab6c-faedea5a7970)

### What did you expect to happen?

Disabling the transform should be the same as deleting it.

### Did this work before?

Yes, we use hosted and it has worked for the past year. only breaking in the last month or two.
Currently running Grafana v10.4.0-65039 (da894994d4)

### How do we reproduce it?

1. Create a working timeseries panel
2. Add transform "Rename fields by regex" 
3. Disable the transform

### Is the bug inside a dashboard panel?

| Key | Value |
|--|--|
| Panel | status-history @ 10.2.0-60477 |
| Grafana | 10.4.0-65039 (da894994d42ee2bde00f2366fe6d30ad00b0e907) // Cloud Pro |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "description": "coarse representation on the versions over time.\n\nOver over points to read the version number.",
      "fieldConfig": {
        "defaults": {
          "custom": {
            "lineWidth": 1,
            "fillOpacity": 70,
            "hideFrom": {
              "tooltip": false,
              "viz": false,
              "legend": false
            }
          },
          "unitScale": true,
          "color": {
            "mode": "continuous-RdYlGr"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "red",
                "value": null
              }
            ]
          },
          "min": 2
        },
        "overrides": []
      },
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "interval": "2h",
      "maxDataPoints": 30,
      "options": {
        "showValue": "auto",
        "rowHeight": 0.9,
        "colWidth": 0.9,
        "legend": {
          "showLegend": true,
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "pluginVersion": "10.2.0-60477",
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "type": "timeseries-multi",
                  "typeVersion": [
                    0,
                    1
                  ],
                  "stats": [
                    {
                      "displayName": "Summary: bytes processed per second",
                      "unit": "Bps",
                      "value": 0
                    },
                    {
                      "displayName": "Summary: lines processed per second",
                      "value": 0
                    },
                    {
                      "displayName": "Summary: total bytes processed",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Summary: total lines processed",
                      "value": 0
                    },
                    {
                      "displayName": "Summary: exec time",
                      "unit": "s",
                      "value": 0.000731358
                    },
                    {
                      "displayName": "Ingester: total reached",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: total chunks matched",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: total batches",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: total lines sent",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: head chunk bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: head chunk lines",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: decompressed bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: decompressed lines",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: compressed bytes",
                      "unit": "decbytes",
                      "value": 0
                    },
                    {
                      "displayName": "Ingester: total duplicates",
                      "value": 0
                    }
                  ],
                  "executedQueryString": "Expr: last_over_time({host=~\"AP.*11.\",units=\"mqtt-node\"} \r\n    |= `geminus` | regexp `geminus.*?1.(?P<minor>\\d+).(?P<version>\\d+)`\r\n    | label_format v=`{{ divf .version 100 | addf .minor | printf \"%1.2f\"}}`\r\n    | v > 1.0\r\n    | unwrap v [30m]) by (host)\r\n\nStep: 2h0m0s",
                  "preferredVisualisationType": "graph",
                  "transformations": [
                    "renameByRegex",
                    "renameByRegex",
                    "renameByRegex"
                  ]
                },
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {
                      "interval": 7200000
                    }
                  },
                  {
                    "name": "Value",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64"
                    },
                    "labels": {
                      "host": "APB-21-01-011A"
                    },
                    "config": {
                      "displayNameFromDS": "APB-21-01-011A"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1705435200000,
                    1705442400000,
                    1705449600000,
                    1705456800000,
                    1705464000000,
                    1705471200000,
                    1705478400000,
                    1705485600000
                  ],
                  [
                    3,
                    3,
                    3,
                    3,
                    3,
                    3,
                    3,
                    3
                  ]
                ]
              }
            },
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "type": "timeseries-multi",
                  "typeVersion": [
                    0,
                    1
                  ],
                  "executedQueryString": "Expr: last_over_time({host=~\"AP.*11.\",units=\"mqtt-node\"} \r\n    |= `geminus` | regexp `geminus.*?1.(?P<minor>\\d+).(?P<version>\\d+)`\r\n    | label_format v=`{{ divf .version 100 | addf .minor | printf \"%1.2f\"}}`\r\n    | v > 1.0\r\n    | unwrap v [30m]) by (host)\r\n\nStep: 2h0m0s",
                  "preferredVisualisationType": "graph",
                  "transformations": [
                    "renameByRegex",
                    "renameByRegex",
                    "renameByRegex"
                  ]
                },
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {
                      "interval": 7200000
                    }
                  },
                  {
                    "name": "Value",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64"
                    },
                    "labels": {
                      "host": "APB-21-01-011B"
                    },
                    "config": {
                      "displayNameFromDS": "APB-21-01-011B"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1705435200000,
                    1705442400000,
                    1705449600000,
                    1705456800000,
                    1705464000000,
                    1705471200000,
                    1705478400000,
                    1705485600000
                  ],
                  [
                    3,
                    3,
                    3,
                    3,
                    3,
                    3,
                    3,
                    3
                  ]
                ]
              }
            }
          ]
        }
      ],
      "title": "Reproduced with embedded data",
      "transformations": [
        {
          "id": "renameByRegex",
          "options": {},
          "disabled": true
        }
      ],
      "type": "status-history"
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >status-history @ 10.2.0-60477</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[loki]</td>\n    </tr>\n    <tr>\n      <th>Transform</th>\n      <td>renameByRegex</td>\n  </tr>\n    <tr><th>Data</th><td> 2 frames, 4 fields, 16 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>10.4.0-65039 (da894994d42ee2bde00f2366fe6d30ad00b0e907) // Cloud Pro</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"datasource\": {\n    \"type\": \"loki\",\n    \"uid\": \"grafanacloud-logs\"\n  },\n  \"description\": \"coarse representation on the versions over time.\\n\\nOver over points to read the version number.\",\n  \"fieldConfig\": {\n    \"defaults\": {\n      \"custom\": {\n        \"lineWidth\": 1,\n        \"fillOpacity\": 70,\n        \"hideFrom\": {\n          \"tooltip\": false,\n          \"viz\": false,\n          \"legend\": false\n        }\n      },\n      \"unitScale\": true,\n      \"color\": {\n        \"mode\": \"continuous-RdYlGr\"\n      },\n      \"mappings\": [],\n      \"thresholds\": {\n        \"mode\": \"absolute\",\n        \"steps\": [\n          {\n            \"color\": \"red\",\n            \"value\": null\n          }\n        ]\n      },\n      \"min\": 2\n    },\n    \"overrides\": []\n  },\n  \"gridPos\": {\n    \"h\": 10,\n    \"w\": 11,\n    \"x\": 0,\n    \"y\": 50\n  },\n  \"id\": 53,\n  \"interval\": \"2h\",\n  \"maxDataPoints\": 30,\n  \"options\": {\n    \"showValue\": \"auto\",\n    \"rowHeight\": 0.9,\n    \"colWidth\": 0.9,\n    \"legend\": {\n      \"showLegend\": true,\n      \"displayMode\": \"list\",\n      \"placement\": \"bottom\"\n    },\n    \"tooltip\": {\n      \"mode\": \"single\",\n      \"sort\": \"none\"\n    }\n  },\n  \"pluginVersion\": \"10.2.0-60477\",\n  \"targets\": [\n    {\n      \"datasource\": {\n        \"type\": \"loki\",\n        \"uid\": \"grafanacloud-logs\"\n      },\n      \"editorMode\": \"code\",\n      \"expr\": \"last_over_time({host=~\\\"${machine:value}\\\",units=\\\"mqtt-node\\\"} \\r\\n    |= `geminus` | regexp `geminus.*?1.(?P<minor>\\\\d+).(?P<version>\\\\d+)`\\r\\n    | label_format v=`{{ divf .version 100 | addf .minor | printf \\\"%1.2f\\\"}}`\\r\\n    | v > 1.0\\r\\n    | unwrap v [${common_interval}]) by (host)\\r\\n\",\n      \"legendFormat\": \"{{host}}\",\n      \"queryType\": \"range\",\n      \"refId\": \"A\",\n      \"step\": \"\"\n    }\n  ],\n  \"title\": \"Versions\",\n  \"transformations\": [\n    {\n      \"id\": \"renameByRegex\",\n      \"options\": {},\n      \"disabled\": true\n    }\n  ],\n  \"type\": \"status-history\"\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above (after transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    },
    {
      "id": 100,
      "title": "Data (before transformations)",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 24,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": false,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: Versions // 2024-01-22 17:41:18",
  "tags": [
    "debug",
    "debug-status-history"
  ],
  "time": {
    "from": "2024-01-16T21:07:41.384Z",
    "to": "2024-01-17T11:20:54.611Z"
  }
}
```
</details>

### Environment (with versions)?

Grafana: 10.4.0-65039
OS:
Browser: Chrome

### Grafana platform?

I use Grafana Cloud

**Comment by usmangt** [comment]
Hi @rpl-will-hamill,

Thanks for opening this issue.

Based on your steps of reproduction, I was also able to reproduce it on my test machine.

I will forward this issue to our Engineers so that they can have a look and analyse it.

**Comment by rovo89** [comment]
Should be fixed by #81320.
